framework/include/utils/MooseUtils.h

```diff
@@ -997,7 +997,7 @@ buildRequiredMaterials(const Consumers & mat_consumers,
  * Note: due to an assertion bug in DynamicStdArrayWrapper we have to allocate one element more
```

Comment:
This is now available in MOOSE: can we modify the need to allocate an extra element?

Reply:
Removed